admin/core/Background/EncryptAndMigrateBackgroundProcess.php

```diff
@@ -54,85 +65,93 @@
 class EncryptAndMigrateBackgroundProcess extends BaseBackgroundProcess
 {
     /**
-     * @var
+     * @var string|null
      */
-    private $passw0rdHash;
+    private ?string $passwordHash = null;
 
     /**
-     * @var null|\Virgil\PureKit\Protocol\Protocol
+     * @var null|CoreProtocol
      */
-    private $protocol;
+    private ?CoreProtocol $protocol;
 
     /**
-     * @var
+     * @var DBQueryHelper|null
      */
-    private $dbqh;
+    private ?DBQueryHelper $dbqh;
 
     /**
-     * @var
+     * @var VirgilCryptoWrapper|null
      */
-    private $vcw;
+    private ?VirgilCryptoWrapper $vcw;
 
     /**
      * @var string
      */
-    protected $action = Config::BACKGROUND_ACTION_MIGRATE;
+    protected string $action = Config::BACKGROUND_ACTION_MIGRATE;
 
-    public function setDep(Protocol $protocol, DBQueryHelper $dbqh, VirgilCryptoWrapper $vcw)
+    /**
+     * @param CoreProtocol $protocol
+     * @param DBQueryHelper $dbqh
+     * @param VirgilCryptoWrapper $vcw
+     * @return void
+     */
+    public function setDep(CoreProtocol $protocol, DBQueryHelper $dbqh, VirgilCryptoWrapper $vcw): void
     {
         $this->protocol = $protocol;
         $this->dbqh = $dbqh;
         $this->vcw = $vcw;
     }
 
     /**
-     * @param mixed $user
-     * @return bool|mixed
-     * @throws \GuzzleHttp\Exception\GuzzleException
-     * @throws \Virgil\PureKit\Exceptions\ProtocolException
+     * @param mixed $item
+     * @return bool
+     * @throws EmptyArgumentException
+     * @throws IllegalStateException
+     * @throws NullArgumentException
+     * @throws PheClientException
+     * @throws PureCryptoException
+     * @throws VirgilCryptoException
      */
-    protected function task($user)
+    protected function task(mixed $item): bool
     {
-        if (is_null($this->passw0rdHash))
-            $this->passw0rdHash = new passw0rdHash();
-
-        $pk = get_option(Option::RECOVERY_PUBLIC_KEY);
-
-        $virgilPublicKey = $this->vcw->importKey(Crypto::PUBLIC_KEY, $pk);
-
-        $encrypted = $this->vcw->encrypt($user->user_pass, $virgilPublicKey);
-
-        $hash = $this->passw0rdHash->get($user->user_pass, 'hash');
-        $params = $this->passw0rdHash->get($user->user_pass, 'params');
-
-        $enrollment = $this->protocol->enrollAccount($hash);
-        $record = base64_encode($enrollment[0]);
-
-        update_user_meta($user->ID, Option::RECORD, $record);
-        update_user_meta($user->ID, Option::PARAMS, $params);
-        update_user_meta($user->ID, Option::ENCRYPTED, $encrypted);
+        try {
+            $this->protocol->getPure()->authenticateUser($item->user_email, $item->user_pass);
```

Comment:
If logic comes to this point, it means, that the plugin has checked the user's database record and defined that the user was not migrated yet. So first action should be "registration" if registration failed, then:
1. It means rare, mostly occurs during developement, when the user was already registered, but database was restored to the state prior to migration.
2. In this case the pluging should try to perform authentication, and if succeed mark this user record  as "migrated".
3. If Authentication failed, it means that user was migrated with another password, and the plugin should offer an option to "change password".